src/main/kotlin/com/autonomousapps/internal/asm.kt

```diff
@@ -587,14 +588,11 @@ private class FieldAnalyzer(
 }
 
 private fun MutableSet<ClassRef>.addClass(classRef: String?, kind: ClassRef.Kind) {
-  classRef?.let {
-    // Strip array indicators
-    it.replace("[", "")
-    // Only add class types (not primitives)
-    if (it.startsWith("L")) {
-      add(ClassRef(it.substring(1, it.length - 1), kind))
-    }
-  }
+  classRef ?: return
+
+  JAVA_FQCN_REGEX_ASM.findAll(classRef)
+    .map { it.value }
+    .forEach { add(ClassRef(it.substring(1, it.length - 1), kind)) }
```

Comment:
Slightly worried this will have other impacts, but if the tests pass, should be good.